pkg/reconciler/testing/v1/eventpolicy.go

```diff
@@ -44,25 +44,69 @@ func NewEventPolicy(name, namespace string, o ...EventPolicyOption) *v1alpha1.Ev
 	return ep
 }
 
-func WithInitEventPolicyConditions(et *v1alpha1.EventPolicy) {
-	et.Status.InitializeConditions()
+func WithInitEventPolicyConditions(ep *v1alpha1.EventPolicy) {
+	ep.Status.InitializeConditions()
+}
+
+func WithTrueAuthenticationEnabledCondition(ep *v1alpha1.EventPolicy) {
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
+			Type:   v1alpha1.EventPolicyConditionAuthnEnabled,
+			Status: corev1.ConditionTrue,
+		})
+}
+
+func WithFalseAuthenticationEnabledCondition(ep *v1alpha1.EventPolicy) {
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
+			Type:   v1alpha1.EventPolicyConditionAuthnEnabled,
+			Status: corev1.ConditionFalse,
+			Reason: "AuthOIDCFeatureNotEnabled",
+		})
+}
+
+func WithTrueSubjectsResolvedCondition(ep *v1alpha1.EventPolicy) {
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
+			Type:   v1alpha1.EventPolicyConditionSubjectsResolved,
+			Status: corev1.ConditionTrue,
+		})
+}
+
+func WithFalseSubjectsResolvedCondition(ep *v1alpha1.EventPolicy) {
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
+			Type:   v1alpha1.EventPolicyConditionSubjectsResolved,
+			Status: corev1.ConditionFalse,
+			Reason: "FromSubjectsNotResolved",
+		})
+}
+
+func WithUnknownSubjectsResolvedCondition(ep *v1alpha1.EventPolicy) {
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
+			Type:   v1alpha1.EventPolicyConditionSubjectsResolved,
+			Status: corev1.ConditionUnknown,
+		})
 }
 
 func WithReadyEventPolicyCondition(ep *v1alpha1.EventPolicy) {
-	ep.Status.Conditions = []apis.Condition{
-		{
+	ep.Status.Conditions = append(ep.Status.Conditions,
+		apis.Condition{
 			Type:   v1alpha1.EventPolicyConditionReady,
 			Status: corev1.ConditionTrue,
-		},
-	}
+		})
 }
 
-func WithUnreadyEventPolicyCondition(ep *v1alpha1.EventPolicy) {
-	ep.Status.Conditions = []apis.Condition{
-		{
-			Type:   v1alpha1.EventPolicyConditionReady,
-			Status: corev1.ConditionFalse,
-		},
+func WithUnreadyEventPolicyCondition(reason, message string) EventPolicyOption {
+	return func(ep *v1alpha1.EventPolicy) {
+		ep.Status.Conditions = append(ep.Status.Conditions,
+			apis.Condition{
+				Type:    v1alpha1.EventPolicyConditionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  reason,
+				Message: message,
+			})
```

Comment:
We could also add this as a 2nd function (e.g. `WithUnreadyEventPolicyConditionWithReason`), so we don't have to update the others and pass empty arguments :shrug:

Reply:
@creydr I actually thought of doing the same earlier but then having a single function makes sense to me for keeping just a single version. Let me know if you think otherwise.

Reply:
I don't have a strong opinion. I just thought that having a `WithUnreadyEventPolicyCondition()` function without arguments might be clearer. But no strong opinion...